lib/chef/resource/macos_userdefaults.rb

```diff
@@ -94,6 +95,7 @@ class MacosUserDefaults < Chef::Resource
 
       property :user, [String, Symbol],
         description: "The system user that the default will be applied to. Set :current for current user, :all for all users or pass a valid username",
+        default: :current,
```

Comment:
it's taken from the default behaviour of `CFPreferencesSetAppValue` https://developer.apple.com/documentation/corefoundation/1515528-cfpreferencessetappvalue that we would want to pick if no host and user is specified.

Reply:
unless `CFPreferencesSetAppValue` represents underpinnings of `defaults` utility or something similar. We should either rely on the behaviour `defaults` utility or fallback to customer preference/experience (from past iteration(s) of `macos_userdefaults`).

Intuitively, I would prefer, `all` instead of `current`, hence, something objective would help make a decision.

Reply:
How the user defaults configuration applies itself is equally dependent on the domain string that we pass. If you look at the defaults utility on the command line, there's no option to set user because it relies on target domain and user elevation. So in order to maintain consistency, it would seem logical to fall back to `:current` user since targeting global or local configuration is done as combinations of `user` property and domain path. 

With regards to your query on past iterations of userdefaults, we had `user` property with no fallback value and a `sudo` flag to elevate permissions. So in a sense, the fallback/default was the local configuration/current user.

Reply:
I dabbled a bit into the old iteration of this resource and indeed, it used to fallback on `current_user` but with a different mechanism than what you have specified here 😄 
This looks good to me, approving!